config/routes.rb

```diff
@@ -0,0 +1,28 @@
+Rails.application.routes.draw do
+  get '/sign_out_user', to: 'users#sign_out_user', as: 'sign_out_user'
+  devise_for :users
+  get '/recipes/:recipe_id/shopping_list', to: 'recipes#shopping_list', as: 'shopping_list'
+  devise_scope :user do
+    authenticated :user do
+      root to: 'foods#index', as: :authenticated_root
+    end
+    unauthenticated :user do
+      root to: 'devise/sessions#new', as: :unauthenticated_root
+    end
+  end
+
+  resources :users, only: [] do
+    resources :foods, only: %i[index show new create destroy]
+    resources :recipes, only: %i[index new create show destroy]
+  end
+
+  resources :recipes, only: %i[index new create show update destroy] do
```

Comment:
[REQUIRED] You are repeating the resources for recipes twice. Kindly use one of them and make it contain the needed nested routes within it.